cmake/Embed.cmake

```diff
@@ -196,7 +196,7 @@ function(add_embed_library EMBED_NAME)
     
     add_library(${EMBED_NAME} INTERFACE)
     if(EMBED_USE_LD)
-        target_sources(${EMBED_NAME} INTERFACE ${OUTPUT_FILES})
+        target_sources(${EMBED_NAME} PRIVATE ${OUTPUT_FILES})
```

Comment:
I dont think this will work. An `INTERFACE` library can only add things to the `INTERFACE`. I thought this would be an error in cmake, but either way if its private then we wont link in the object files so there should be linking errors.